### PR TITLE
feat: allow templates in annotations

### LINKED ISSUE #254: Allow using templates in annotations
I have a use case where I need to include helm templates in an annotation. Example:

```yaml
  ingress: 
    enabled: true
    annotations:
      nginx.ingress.kubernetes.io/custom-http-errors: "401,403,404,500,502,503"
      nginx.ingress.kubernetes.io/default-backend: "{{ .Release.Name }}-error-pages"
```

This does not work as of now - the "{{ .Release.Name }}"-part is not interpreted and written as is to the rendered manifest.

### PR DESCRIPTION
**Pull request by gtudan**
This allows templates in annotations for all resources. It tried to keep the pattern used in other places (i.e. extra-env for deployment) and use the same pattern for all annotations.

/fixes #254

### CODE REVIEW

File: helm/oauth2-proxy/ci/tpl-values.yaml [hunk 1]
```diff
@@ -8,6 +8,8 @@ ingress:
   tls:
     - hosts:
         - "{{ $.Release.Name }}.local"
+  annotations:
+    nginx.ingress.kubernetes.io/default-backend: "{{ $.Release.Name }}-default-backend"
```
Comment (pierluigilenoci) [inline helm/oauth2-proxy/ci/tpl-values.yaml:12]: Why this? You don't need to add this.

File: helm/oauth2-proxy/templates/secret-alpha.yaml [hunk 2]
```diff
@@ -7,8 +7,8 @@
 apiVersion: v1
 kind: Secret
 metadata:
-{{- if .Values.alphaConfig.annotations }}
-  annotations: {{- toYaml .Values.alphaConfig.annotations | nindent 4 }}
+{{- with .Values.alphaConfig.annotations }}
+{{ tpl ( toYaml . ) $ | indent 4 }}
```
Comment (pierluigilenoci) [inline helm/oauth2-proxy/templates/secret-alpha.yaml:11]: Can you put it back?
```
annotations:
```

**Review by pierluigilenoci — CHANGES_REQUESTED** [review]
There are a few things to fix. 
Also missing are the chart version bump and release notes for ArtifactHub.

**Comment by pierluigilenoci** [comment]
I approved the tests; at least you can see which tests failed.

File: helm/oauth2-proxy/templates/secret-alpha.yaml (hunk 2 above)
Comment (gtudan) [reply helm/oauth2-proxy/templates/secret-alpha.yaml:12]: Sorry, copy-pasted to much - it's back, thanks for noticing!

File: helm/oauth2-proxy/ci/tpl-values.yaml (hunk 1 above)
Comment (gtudan) [reply helm/oauth2-proxy/ci/tpl-values.yaml:12]: I have to confess I don't fully understand how the tests work - I added this so I could see if the template is being resolved and to ensure the linter won't complain about the format.

If you can give me a hint on where to add a separate test case for this, then I'm happy to do so :-)

**Comment by gtudan** [comment]
>  Also missing are the chart version bump and release notes for ArtifactHub.

I incremented the chart version and added the changes as annotations - hope I got the format right?

Sorry, I checked for a contributing instruction, but didn't find one...

**Comment by gtudan** [comment]
@pierluigilenoci  The last merge introduced an extra newline and the linter complained about it. I removed it and pushed again. Could you please approve the workflow again?

**Comment by tuunit** [comment]
Approved the run

File: helm/oauth2-proxy/ci/tpl-values.yaml (hunk 1 above)
Comment (pierluigilenoci) [reply helm/oauth2-proxy/ci/tpl-values.yaml]: @gtudan to test the chart we use ct.
Specifically this https://github.com/helm/chart-testing/blob/main/doc/ct_lint.md
> Charts may have multiple custom values files matching the glob pattern '*-values.yaml' in a directory named 'ci' in the root of the chart's directory. The chart is linted for each of these files. If no custom values file is present, the chart is linted with defaults.

**Review by pierluigilenoci — APPROVED** [review]

File: helm/oauth2-proxy/ci/tpl-values.yaml (hunk 1 above)
Comment (pierluigilenoci) [reply helm/oauth2-proxy/ci/tpl-values.yaml]: @gtudan, I have already approved the PR, but if you want to add to the tests all the cases you added in the PR, that would be great, so we have them all covered.

Comment (pierluigilenoci) [reply helm/oauth2-proxy/ci/tpl-values.yaml]: Ok, I did it for you. 😄

**Comment by pierluigilenoci** [comment]
> I incremented the chart version and added the changes as annotations - hope I got the format right?

You did well 😄 

> Sorry, I checked for a contributing instruction, but didn't find one...

In reality, a contribution guide is missing.

**Review by tuunit — APPROVED** [review]

**Pull request merged**